app/actions/remote/role.ts

```diff
@@ -43,11 +43,13 @@ export const fetchRolesIfNeeded = async (serverUrl: string, updatedRoles: string
     try {
         const roles = await client.getRolesByNames(newRoles);
 
-        if (roles.length) {
-            await operator.handleRole({
-                roles,
-                prepareRecordsOnly: false,
-            });
+        if (!fetchOnly) {
+            if (roles.length) {
```

Comment:
You can put these two ifs in one single if. (i.e. `if (!fetchOnly && roles.length)`)

Reply:
done.  I also modified one other area in the same commit.